Utils.py

```diff
@@ -1026,3 +1026,30 @@ def is_iterable_except_str(obj: object) -> TypeGuard[typing.Iterable[typing.Any]
     if isinstance(obj, str):
         return False
     return isinstance(obj, typing.Iterable)
+
+
+def get_full_typename(t: type) -> str:
+    if module := getattr(t, "__module__"):
```

Comment:
What's the difference between `getattr(t, "__module__")` and `t.__module__`?
Did you think it was doing `getattr(t, "__module__", None)`?
Did you find types that don't have `__module__`?

Reply:
Changed to `t.__module__`.
I think I was using `getattr` based on some style I found elsewhere in AP but now I don't remember, and maybe I did find exceptions that don't have __module__ (per https://github.com/python/cpython/issues/120857 it is possible) but I'm not reproducing it now so maybe not.
VSCode told me that `getattr` default was `None` and I forgot that getattr is actually overloaded to tell the difference.